programmers/신규-아이디-추천.js

```diff
@@ -0,0 +1,20 @@
+function solution(new_id) {
+    new_id = new_id.toLowerCase(); //1단계
+    
+    new_id = new_id.replace(/[^a-z0-9-_.]/g, "" ); // 2단계
+    
+    new_id = new_id.replace(/\.{2,}/g, "." ); //3단계
+    
+    new_id = new_id.replace(/^\.|\.$/g, "" );  //4단계
+    
+    if (new_id.length === 0) new_id = "a";  //5단계
+    
+    new_id = new_id.slice(0, 15).replace(/\.$/, "" );  //6단계
```

Comment:
하나 하나 다 new_id 에 할당하는 것 보다 이렇게 이어서 쓸 수 있음 
그리고 굳이 new_id.length === 0 을 if문으로 확인하기 보다 `.replace(/^$/, 'a')` 을 넣으면 됨
```suggestion
     new_id = new_id
        .toLowerCase() //1단계
        .replace(/[^\w\.\-]/g, '') //2단계
        .replace(/[\.]{2,}/g, '.') //3단계
        .replace(/^\./, '') //4단계 맨 앞 '.' 제거
        .replace(/\.$/, '') //4단계 맨 뒤 '.' 제거
        .replace(/^$/, 'a') //5단계
        .slice(0, 15).replace(/\.$/, ''); //6단계
```